memory available and max-mem pod schedulable capacity checks

joelddiaz opened the pull request.

Commit:
memory available and max-mem pod schedulable capacity checks
plus new 'conversions' library
and necessary RPM spec updates to include capacity checks into monitoring-openshift.rpm

twiest commented on scripts/monitoring/cron-send-cluster-capacity.py:
```diff
@@ -0,0 +1,292 @@
+        return cpu_num
+
+    @staticmethod
+    def byte_me(num):
```
should be `convert_to_byte`.

Should also be in a library somewhere.

twiest commented on scripts/monitoring/cron-send-cluster-capacity.py:
```diff
@@ -0,0 +1,292 @@
+        self.zagg_sender.add_zabbix_keys({zbx_key_max_pods: schedulable})
+
+    @staticmethod
+    def cpu_convert(num):
```
should be in a library somewhere.

twiest commented on scripts/monitoring/cron-send-cluster-capacity.py:
```diff
@@ -0,0 +1,292 @@
+                                      debug=self.args.debug)
+
+        master_cfg = []
+        with open('/etc/origin/master/master-config.yaml', 'r') as yml:
```
Should be able to pass this in. Should default to it, though.

NO MAGIC NUMBERS (or strings)

twiest commented on scripts/monitoring/cron-send-cluster-capacity.py:
```diff
@@ -0,0 +1,292 @@
+                            default=None, help='Verbose?')
+        parser.add_argument('--debug', action='store_true',
+                            default=None, help='Debug?')
+        parser.add_argument('--test', action='store_true', default=False,
```
We've been using `--dry-run` for this. Would be nice to stay consistent.

twiest commented on scripts/monitoring/cron-send-cluster-capacity.py:
```diff
@@ -0,0 +1,292 @@
+                                           self.byte_me(mem)
+            else:
+                # no limit defined for container
+                pass
```
Why have an else clause at all?

twiest commented on scripts/monitoring/cron-send-cluster-capacity.py:
```diff
@@ -0,0 +1,292 @@
+                                             self.byte_me(mem)
+            else:
+                # no limit defined for container
+                pass
```
Why have an else clause at all?

joelddiaz commented:
@twiest: split out the cpu and byte conversions into python-openshift-tools. you can now pass in your own master-node.yaml location. converted to --dry-run. removed useless 'else' clauses.

twiest commented:
:+1:
